### PR TITLE
RxDB Refactor

### PR DESCRIPTION
**Pull request by JFKFred**
* Refactor the RxDB structure.
* Consolidate from 5 collections to 3 collections.
* Utilize database functionality instead of relying on local storage.
* Remove unnecessary hooks resulting from outdated and inefficient structuring.
* Status is now directly linked with the asset

**Database structure**
* MetadataCollection: { assetName, metadata, status }
* ValidationsCollection: { assetName, validation }
* ProjectCollection: { name, nfts, unchecked, errorFlagged, errorDetected, valids }

### CODE REVIEW

File: client/next-env.d.ts [hunk 1]
```diff
@@ -2,4 +2,4 @@
 /// <reference types="next/image-types/global" />
 
 // NOTE: This file should not be edited
-// see https://nextjs.org/docs/basic-features/typescript for more information.
+// see https://nextjs.org/docs/app/building-your-application/configuring/typescript for more information.
```
Comment (pChausseC) [inline client/next-env.d.ts:5]: this file should not be edited?

File: client/src/providers/rxdb.provider.tsx [hunk 2]
```diff
@@ -1,25 +1,55 @@
...
+    if (db) {
+      await db.remove(); // Remove the existing database
+    }
+    const newDb = await initialize(); // Create a new instance of the database
+    setDb(newDb); // Update the state with the new database
+  };
+
+  useEffect(() => {
+    // Initialize the database on component mount
+    reinitializeRxDB().catch(console.error);
```
Comment (pChausseC) [inline client/src/providers/rxdb.provider.tsx]: db gets removed and re initialized on every page load (im not sure thats efficient)

**Review by pChausseC — CHANGES_REQUESTED** [review]

File: client/src/providers/rxdb.provider.tsx (hunk 2 above)
Comment (pChausseC) [reply client/src/providers/rxdb.provider.tsx]: `reinitializeRxDB` removes the db, but the db should already have been removed by the clean up function of the useEffect

**Review by pChausseC — CHANGES_REQUESTED** [review]

File: client/next-env.d.ts (hunk 1 above)
Comment (JFKFred) [reply client/next-env.d.ts:5]: i've run npm update, this might be why this have change ?

**Review by pChausseC — APPROVED** [review]

**Pull request merged**